core/deployment/src/main/java/io/quarkus/deployment/configuration/tracker/ConfigTrackingConfig.java

```diff
@@ -0,0 +1,83 @@
+package io.quarkus.deployment.configuration.tracker;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.quarkus.util.GlobUtil;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
+
+/**
+ * Configuration options for application build time configuration usage tracking
+ * and dumping.
+ */
+@ConfigMapping(prefix = "quarkus.config-tracking")
+@ConfigRoot(phase = ConfigPhase.BUILD_TIME)
+public interface ConfigTrackingConfig {
+
+    /**
+     * Whether configuration dumping is enabled
+     */
+    @WithDefault("false")
+    boolean enabled();
+
+    /**
+     * Directory in which the configuration dump should be stored.
+     * If not configured the {@code .quarkus} directory under the project directory will be used.
+     */
+    Optional<Path> directory();
+
+    /**
+     * File to which the configuration should be dumped. If not configured, the {@link #filePrefix} and
+     * {@link #fileSuffix} will be used to generate the final file name.
+     * If the configured file path is absolute, the {@link #directory} option will be ignored. Otherwise,
+     * the path will be considered relative to the {@link #directory}.
+     */
+    Optional<Path> file();
+
+    /**
+     * File name prefix. This option will be ignored in case {@link #file} is configured.
+     */
+    @WithDefault("quarkus")
+    String filePrefix();
+
+    /**
+     * File name suffix. This option will be ignored in case {@link #file} is configured.
+     */
+    @WithDefault("-config-dump")
+    String fileSuffix();
+
+    /**
+     * A list of config properties that should be excluded from the report.
+     * GLOB patterns could be used instead of property names.
+     */
+    Optional<List<String>> exclude();
+
+    /**
+     * Translates the value of {@link #exclude} to a list of {@link java.util.regex.Pattern}.
+     *
+     * @return list of patterns created from {@link #exclude}
+     */
+    default List<Pattern> getExcludePatterns() {
+        if (exclude().isEmpty()) {
+            return List.of();
+        }
+        var list = exclude().get();
+        final List<Pattern> excludePatterns = new ArrayList<>(list.size());
+        for (var s : list) {
+            excludePatterns.add(Pattern.compile(GlobUtil.toRegexPattern(s)));
+        }
+        return excludePatterns;
+    }
+
+    /**
+     * Whether to use a {@code ~} as an alias for user home directory in path values
+     */
+    @WithDefault("true")
+    boolean useUserHomeAliasInPaths();
```

Comment:
this seems out of place. 

1) why have this as an option in this specific corner of Quarkus ?
2) why do we need to refer to user home path anyways for this - is it not supposed to be relative to the project ?

Reply:
ah sorry I read it as this was used for reading variables.

..so I grok the relevance but is there even a value in storing this as afaik we don't interpret ~ into ${user.home} when reading the values ?

Reply:
I haven't tried using `~` in config values, tbh. If we don't do this kind of transformation, every config will be different for each user. It will often still be even with `~` but user home dir path looks like an obvious thing to ignore in comparisons.